### PR TITLE
ref(dx): Rename `--crons` -> `--celery-beat`

### PR DESCRIPTION
**Pull request by evanpurkhiser**
We have a product nanemd `crons` now. This should help us avoid
conflating these two things.

### CODE REVIEW

File: src/sentry/runner/commands/devserver.py [hunk 1]
```diff
@@ -257,14 +257,14 @@ def devserver(
         click.echo("--ingest was provided, implicitly enabling --workers")
         workers = True
 
-    if workers and not crons:
+    if workers and not celery_beat:
         click.secho(
-            "If you want to run celery crons (celerybeat workers), you need to also pass --crons.",
+            "If you want to run celery crons (celerybeat workers), you need to also pass --celery-beat.",
```
Comment (markstory) [inline src/sentry/runner/commands/devserver.py:262]:
```suggestion
            "If you want to run periodic tasks from celery (celerybeat), you need to also pass --celery-beat.",
```

🤷

**Review by markstory — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
Going to let @asottile-sentry take a look as well, not sure if there's any gotchas here.

**Review by asottile-sentry — APPROVED** [review]
seems fine -- please update the documentation as well (if applicable) in notion and sentry dev docs, announce in #discuss-backend, #discuss-dev-infra, #team-engineering, and send out a notice in vanguard probably as well?

**Comment by evanpurkhiser** [comment]
Sounds good.

Couldn't find anywhere this flag is actually documented, it was added relatively recently [e949874c3e00ca3dd69e930022137e045b4a34fc](https://github.com/getsentry/sentry/pull/53817)

**Pull request merged**